test/exe/main_common_test.cc

```diff
@@ -117,12 +117,32 @@ TEST_P(MainCommonTest, ConstructWritesBasePathId) {
 #endif
 }
 
-// Exercise enabling core dump.
+// Exercise enabling core dump and succeeding.
+// Note: this test will call the real prctl(), which is what we want.
 TEST_P(MainCommonTest, EnableCoreDump) {
   addArg("--enable-core-dump");
   VERBOSE_EXPECT_NO_THROW(MainCommon main_common(argc(), argv()));
 }
 
+class MockPlatformImpl : public PlatformImpl {
+  bool enableCoreDump() override { return false; }
+}
+
+// Exercise enabling core dump and failing.
+// Note: this simulates a failed prctl() call.
+TEST_P(MainCommonTest, EnableCoreDumpFails) {
+  Event::TestRealTimeSystem real_time_system;
+  DefaultListenerHooks default_listener_hooks;
+  ProdComponentFactory prod_component_factory;
+
+  const auto args = std::vector<std::string>(
+      {"envoy-static", "--use-dynamic-base-id", "-c", config_file_.c_str(), "--enable-core-dump"});
+  OptionsImpl options(args, &MainCommon::hotRestartVersion, spdlog::level::info);
+  MainCommonBase first(options, real_time_system, default_listener_hooks, prod_component_factory,
```

Comment:
Can you actually expect the call and then also use EXPECT_LOG_CONTAINS to check for the warning?

Reply:
done